src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/PrimitiveResolverTest.cs

```diff
@@ -69,9 +67,9 @@ public void PrimitiveTest2()
                 var bin = MessagePackSerializer.Serialize<object>(x);
                 var re1 = (object[])MessagePackSerializer.Deserialize<object>(bin);
 
-                x[0].Is((int)(byte)re1[0]);
-                x[1].Is((int)(byte)re1[1]);
-                x[2].Is((int)(ushort)re1[2]);
+                x[0].Is((int)re1[0]);
+                x[1].Is((int)re1[1]);
+                x[2].Is((int)re1[2]);
```

Comment:
I feel like we should add tests that mix it up -- include `byte` and `ushort` values to see that those too get preserved -- not just `int`s.  Perhaps `float` vs. `double` as well.

Reply:
This is just trying to get existing broken tests to work, so there is no new scenario that would cover a lot of ground.
float, double preserves the type from the original (it doesn't shrink the double to a float), though, so I'm sure it's fine.